### DISCUSSION TITLE
Anyone interested in digging through the Forks?

### DISCUSSION THREAD

**Discussion by ericblade** [ericblade/quagga2#318] [General]
There's over 600 forks of the original repo, and I think most anyone would be surprised if more than half had any commits in them, but the ones that do might have some very good things that were never PR'd to the original repo.  Anyone know of any scripts that are good to dig through that sort of thing?  Or interested in doing it?  (i mean, i am interested in doing it, but if there's anyone else that wants to help, maybe we can split the task up some)

**Comment by xigolle** [comment]
@ericblade I might be interested in helping out with this task. 
I could look through all the forks and see if there additional commits and list all of them that do. 
Although to see if the commits are relevant for the project I might need some help. 

I am using the package but haven't dive in the code itself. (Will be doing the task I mentioned over the upcoming weekend)

**Comment by ericblade** [comment]
Of course we'd probably run into a bunch of things that aren't necessarily relevant to the project, customizations that don't make sense to pull in, but it can't hurt to look, and ask people to PR to here, and make notes of anything.  If you want to start at one side of the list or the other, make note where you stop, and I or someone else can pick up where you leave off :-)

it'd be really nice if there were some script to dig through all the forks and output which ones have commits that aren't in a particular fork.

**Comment by ericblade** [comment]
I dug through the end of the list from "t" to "z" by username, and noted a few interesting looking commits in new issues.

**Comment by TomasHubelbauer** [comment]
I did usernames starting N through S. These are the ones where I found ahead commits in master. I also looked at branches of each, but to my quick glance none so far looked like they had new forker-initiated branches. They all have different branches based on when they were forked but the names of the branches all look like they come from the original repository.

- https://github.com/Swapcard-Corporation/quaggaJS/commits/master TypeScript typings
- https://github.com/stianstr/quaggaJS/commits/master pass video element in directly
- https://github.com/Shipmetrics/quaggaJS/commits/master random fixes?
- https://github.com/sharikovvladislav/quaggaJS/commits/master typings fix
- https://github.com/sascha-hendel/quaggaJS/commits/master media devices update
- https://github.com/Rosataker/quaggaJS/commits/master live_w_locator.js (31 commits!)
- https://github.com/qhkong/quaggaJS/commits/master
- https://github.com/nikytwo/quaggaJS/commits/master reduce false positives
- https://github.com/nickperez1285/quaggaJS/commits/master ???
- https://github.com/NeraiCorp/getUserBarcode/commits/master readme and bower
- https://github.com/n-psk/quaggaJS/commits/master whitesource thingy

---

Also, as a bit of an unsolicited "advice"/suggestion, you might want to recreate this repo from scratch and push your fork to it instead of forking Christoph's one on GitHub so it doesn't show the GitHub parent repo association.

It IMO makes a repo looks more legitimate, not lost in the sea of 700 other forks there already are and as we're seeing all abandoned. You will lose watchers and stars you have so far but maybe better to decide to do this now then later. Also with the for now small amount of starrers and watchers you can @-mention them to let them know to restar/rewatch. If you preserve the name links you have posted so far will keep working.

I am saying this because you seem determined to actually take this somewhere and not abandon it and it would be a shame to not have this repository stand out and communicate that.

**Comment by ericblade** [comment]
Thanks for those!

 I clicked a few, and saw some interesting updates (the video element, and the random fixes one particularly)

It seems that a fork can be de-associated by github support, according to https://help.github.com/en/articles/why-are-my-contributions-not-showing-up-on-my-profile#commit-was-made-in-a-fork .. i hadn't considered that before, but might be something to do after we've picked through the rest of the network

**Comment by TomasHubelbauer** [comment]
M:

- https://github.com/musatov/quaggaJS/commits/master container for video element

L:

- https://github.com/Louvet/quaggaJS/commits/master QR fixes

K:

- https://github.com/KazukiNoto/quaggaJS/commits/master demo and locator changes
- https://github.com/karl-sjogren/quaggaJS/commits/master plessey reader

J:

- https://github.com/jtn-d/quaggaJS/commits/master initWithVideo
- https://github.com/jdboyce/quaggaJS/commits/master layout stuff
- https://github.com/JauernigIT/quaggaJS branches!
- https://github.com/ja-mescher/quaggaJS/commits/master readme

I:

- https://github.com/imanderson/quaggaJS/commits/master optional autostart

H: nothing

G:

- https://github.com/gigarza/quaggaJS/commits/master 12 commits
- https://github.com/groy/quaggaJS/commits/master ditch lodash

F: nothing

E: nothing

D:

- https://github.com/denisoby/quaggaJS/commits/master decodeSingle
- https://github.com/Domratchev/quaggaJS/commits/master TypeScript port!

C:

- https://github.com/Cosium/quaggaJS/commits/bugfix/cannot-pass-offset-to-detection-area maybe
- https://github.com/cbo2/quaggaJS/commits/master changes 🤷‍♂
- https://github.com/cambirch/quaggaJS/commits/master decoding stuff

B:

- https://github.com/Bajdzis/quaggaJS/commits/optimization-import

A:

- https://github.com/ARDonley/quaggaJS/commits/master try catch / exceptions
- https://github.com/apsmcc/quaggaJS/commits/master format
- https://github.com/agusdutra/quaggaJS/commits/master readme
- https://github.com/ad-m/quaggaJS/commits/parallel-process fix parallel

That should be all. :-)

**Comment by ericblade** [comment]
Thanks so much @TomasHubelbauer i'll pick through the finds and see if I can reach out to those devs and/or file an issue here to investigate some of them further!

**Comment by ericblade** [comment]
I'm hoping that @domratchev will be interested in contributing his typescript changes, but with better organization, otherwise I feel like I'm going to use that as a base for handling it on my own sometime, so i'm going to skip over the typescript related changes

@stianstr https://github.com/stianstr/quaggaJS/commit/b2c716724cb634ca89e0526305acedca0a0838e8 adds ability to pass in a pre-constructed video element.  i feel like this is probably a good idea, though i'm not sure exactly why

@jtn-d does similar, but maybe more fleshed out? here https://github.com/jtn-d/quaggaJS/commit/cb0b779cb87a3da4e8b0243bd211f36d580e58a2

@musatov does the same thing, but differently: https://github.com/musatov/quaggaJS/commit/250e9038e436ad12336b5849838c0b80d6318a6f

I think that means it's probably a desireable change, though with three wildly different implementations, we should take a few minutes to try to grasp their goals

@shipmetrics has fixes that look good at a quick glance, https://github.com/Shipmetrics/quaggaJS/commits/master  .. the test for firefox seems questionable, but otherwise, it enhances the resolution of devices and supported constraints and facingMode . i'm not sure ultimately what the effects are but the change *looks* sound.

@sascha-hendel also contributes some potential enhancements to getUserMedia: https://github.com/sascha-hendel/quaggaJS/commit/2e33ed5437d28f5e6fc5f082617950d17dc6ec2d

Despite Rosataker having 31  commits, it's all just minor customizations

@nikytwo makes code_128 more restrictive, but apparently also generates more failures despite reducing number of false positives? we might want to tweak code_128's error tolerance down a little bit, but maybe not that much. https://github.com/nikytwo/quaggaJS/commit/ca667db2901a6c74a069bad48f78039065d90f2e

@ad-m parallel processing fix probably makes sense, but is an interface change. https://github.com/ad-m/quaggaJS/commits/parallel-process  There may be a way there without changing the interface, I'm not sure.

@cambirch's top 4 commits here look really interesting https://github.com/cambirch/quaggaJS/commits/master i'm definitely going to reach out and ask them to PR, or pull them myself if i don't hear anything

@cosium's "cannot pass offset" looks interesting, though i'm not sure if offset is a feature already that is broken, or if it's a new feature. either way, looks interesting. https://github.com/Cosium/quaggaJS/commit/49b256d005d3473fed9659bcdc51a2d025f5fcb2

@denisoby has an interesting example showing how to use pdfjs to read barcodes from PDFs, I think? https://github.com/denisoby/quaggaJS/tree/master/example/pdfjs  doesn't look complete, though, I could be wrong

@imanderson's change for not auto-starting when given an inputstream is probably good? https://github.com/imanderson/quaggaJS/commit/07c2f51443b2bfd0048c72ea444decefc7abd7e3

@karl-sjogren adds a new reader for "plessey" type barcodes

@Louvet adds qr code reading, i'll note it in the other ticket on QR codes.

**Comment by denisoby** [comment]
> @denisoby has an interesting example showing how to use pdfjs to read barcodes from PDFs, I think? https://github.com/denisoby/quaggaJS/tree/master/example/pdfjs doesn't look complete, though, I could be wrong

Hi guys. It was pretty long ago, but if you are interested in it, I can try to recall what I did there almost 2 years ago and compile a complete PR. With probably example of usage quagga and pdfjs together.

**Comment by ericblade** [comment]
@denisoby although I don't have any practical use for it, i'm all for more example usages :-) I would totally welcome it

**Comment by karl-sjogren** [comment]
I always had the intention to make a PR of the plessey reader  but I never got it working well enough for it. Not even sure if it is in a working state at all right now so I'll just add a notice in my repo that it is still WIP.

**Comment by xigolle** [comment]
Really nice you already went to the list @TomasHubelbauer :) If there is anything I can help with @ericblade  let me know! :) Not the most experienced developer but most certain want to help ;)

**Comment by ericblade** [comment]
The various forks that implement passing in a pre-constructed video element apparently failed to notice that that's already possible, although I think I want to investigate their methods some.  The methods for passing in preconstructed html elements are in desperate need of being updated to take actual element references rather than doing querySelector all over the place, so they don't force your html to have a specific structure that may or may not make sense (see #33 for a bit of explanation, and an example of what I think should be avoided)

**Comment by ericblade** [comment]
Now that we've gone through the list, I submitted a request to support to de-parent the fork, and now we're a separate repo, and hopefully pull reqs will default to this repo master, instead of the original repo master :)

**Comment by TomasHubelbauer** [comment]
And I see the request had been granted, awesome!
